src/autopas/containers/hierarchicalGrids/HierarchicalGrids.h

```diff
@@ -246,212 +215,130 @@ class HierarchicalGrids : public ParticleContainerInterface<Particle> {
-        const bool smallerLevelEmpty = _hierarchyLevels[smallerLevel].size() == 0;
-        if (largerLevelEmpty or smallerLevelEmpty){
-          continue;
-        }
-        
-        
-        const double crossLevelCellSize = ( _hierarchyLevels[largerLevel].getCutoff() + 
-                                            _hierarchyLevels[smallerLevel].getCutoff() ) / 2.0;
-
-        //Create cross-level
-        autopas::LinkedCells<Particle> crossLevel(_boxMin, _boxMax, crossLevelCellSize, _skinPerTimestep, _rebuildFrequency, 
-                                                  _cellSizeFactor, _loadEstimator);
-        
-        //Very ugly trick: Make particle radii of the larger level negative
-        //Add large particles
-        for(auto iterLarger = _hierarchyLevels[largerLevel].begin(); iterLarger.isValid(); ++iterLarger) {
-          const double particleRad = - 1.0 * iterLarger->getRad();
-          iterLarger->setRad(particleRad);
-          crossLevel.addParticleImpl(*iterLarger);
-          }
-        _hierarchyLevels[largerLevel].deleteAllParticles();
-        
-
-        //Add small particles
-        for(auto iterSmaller = _hierarchyLevels[smallerLevel].begin(); iterSmaller.isValid(); ++iterSmaller) {
-          crossLevel.addParticleImpl(*iterSmaller);
-        }
-        _hierarchyLevels[smallerLevel].deleteAllParticles();
-
-        // Iterate pairwise over the cross-level
-        crossLevel.iteratePairwise(traversal);
-
-        // Move particles back to their level
-        AUTOPAS_OPENMP(parallel)
-        for (auto iterCross = crossLevel.begin(); iterCross.isValid(); ++iterCross) {
-
-          const double particleRad = iterCross->getRad();
-
-          //if targetLevel = largerLevel, particle has negative radius. Hence, reverse the ugly trick
-          if (particleRad < 0.) {
-            
-            iterCross->setRad(std::abs(particleRad));
-          }
+  void iterateAllCrossLevels() {
+    AUTOPAS_OPENMP(parallel)
+    for (auto &levelPairs : _crossLevelInteractions) {
+      size_t largerLevel, smallerLevel;
+      std::tie(largerLevel, smallerLevel) = levelPairs;
 
-          unsigned int targetLevel = getHierarchyLevelOfParticle(*iterCross);
-
-          addParticleToGridLevel(*iterCross, targetLevel);
-        }
-        
-        crossLevel.deleteAllParticles();
+      iterateOverCrossLevels(largerLevel, smallerLevel);
+    }
+  }
 
+  void iterateOverCrossLevels(unsigned int firstLevel, unsigned int secondLevel) {
+    demLib::DEMFunctor<Particle> demFunctor(1.);
+
+    // iterate over all particles in a given larger level
+    //  @note Maybe parallelize iteration over second level?
+    // AUTOPAS_OPENMP(parallel firstprivate(demFunctor)) // <- cannot parallelize here, since this leads to erroneous
+    // forces
+    for (auto iterFirstLVLParticles = _hierarchyLevels[firstLevel].begin(); iterFirstLVLParticles.isValid();
+         ++iterFirstLVLParticles) {
```

Comment:
```suggestion
    for (const auto particle : _hierarchyLevels[firstLevel]) {
```
Not sure but I think this works. (And then change all subsequent `iterFirstLVLParticles->` to `particle.`)

Reply:
Not with const, since I otherwise get a 'binding reference of type ‘demLib::DEMParticle&’ to ‘const demLib::DEMParticle’ discards qualifiers' error